lib/faraday/utils.rb

```diff
@@ -95,7 +95,14 @@ def to_hash() ::Hash.new.update(self) end
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Find the last set of response headers.
+        start_index = headers.rindex { |x| x.match(/^HTTP\//) }
```

Comment:
@iMacTia I think `headers.rindex { |x| x.match(/^HTTP\//) }.to_i` should fix the problem (but I didn't test it).
If I understand correctly, this would simply cause it to start from the top, if `HTTP` header is missing. It's not VCR specific as I believe faraday shouldn't fail in this place anyway (the change is basically backwards incompatible by assuming the server response will always contain a certain string).
Looking at VCR code I now believe this was a conscious decision so it's even less likely to be fixed.